scripts/test_run_lifecycle_soak.py

```
@@ -0,0 +1,455 @@
+        pass
+
+
+def test_restoration_guard_preserves_an_unexpected_primary_and_prints_residue() -> None:
+    primary = RuntimeError("proof driver crashed")
+    stderr = io.StringIO()
+    try:
+        with redirect_stderr(stderr), MODULE.restoration_guard(lambda: [f"{KEY}: Access is denied"]):
+            raise primary
+    except RuntimeError as reported:
+        if reported is not primary:
+            raise AssertionError("restoration replaced the unexpected primary exception")
```

Ruff (0.16.1)

[warning] Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)